src/tests/ftest/util/launch_utils.py

```diff
@@ -77,6 +78,64 @@ def setup_fuse_config(logger, hosts):
             raise LaunchException(f"Failed to setup {config}")
 
 
+def __add_systemctl_override(logger, hosts, service, user, command, config, path, lib_path):
+    """Add a systemctl override file for the specified service.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        hosts (NodeSet): hosts on which to create the systemctl config
+        service (str): service for which to issue the command
+        user (str): user to use to issue the command
+        service_command (str): full path to the service command
+        service_config (str): full path to the service config
+        path (str): the PATH variable to set in the systemd config.
+        ld_library_path (str): the LD_LIBRARY_PATH variable to set in the systemd config.
+
+    Raises:
+        LaunchException: if setup fails
+
+    Returns:
+        dict: a dictionary of the systemctl override config file key with a dictionary value
+            containing the related host and user information
+    """
+    logger.debug("-" * 80)
+    logger.info("Setting up systemctl override for %s", service)
+    try:
+        systemctl_override = create_override_config(
+            logger, hosts, service, user, command, config, path, lib_path)
+    except SystemctlFailure as error:
+        raise LaunchException(f"Failed to setup systemctl config for {service}") from error
+    return {systemctl_override: {"hosts": hosts, "user": user}}
+
+
+def setup_systemctl(logger, servers, clients, test_env):
+    """Set up the systemctl override files for the daos_server and daos_agent.
+
+    Args:
+        logger (Logger): logger for the messages produced by this method
+        servers (NodeSet): hosts that may run the daos_server command
+        clients (NodeSet): hosts that may run the daos_agent command
+        test_env (TestEnvironment): the test environment
+
+    Raises:
+        LaunchException: if setup fails
+
+    Returns:
+        dict: a dictionary of systemctl override config file keys with NodeSet values identifying
+            the hosts on which to remove the config files at the end of testing
+    """
+    systemctl_configs = {}
+    systemctl_configs.update(
+        __add_systemctl_override(
+            logger, servers, "daos_server.service", "root", "/usr/bin/daos_server",
+            test_env.server_config, None, None))
+    systemctl_configs.update(
+        __add_systemctl_override(
+            logger, clients, "daos_agent.service", test_env.agent_user, "/usr/bin/daos_agent",
+            test_env.client_config, None, None))
+    return systemctl_configs
```

Comment:
So this just hardocdes the `/usr` binaries for now? This makes it impossible to run out of a source build without code changes. E.g. on my boro node I manually modify the systemd config to point to my source build and now this will always clobber that config to be incorrect.

We need to get the correct binaries from `self.prefix`

Reply:
I broke this into two parts:
1) This PR to verify functionality while still using root/sudo/RPMs
2) DAOS-16220 to follow which will support specifying an alternate user for daos_agent, including optional PATH and LD_LIBRARY_PATH settings.

Reply:
Thanks. Figured :)